packages/builder/src/pages/builder/portal/apps/_FirstAppOnboarding/index.svelte

```diff
@@ -0,0 +1,198 @@
+<script>
+  import { goto } from "@roxi/routify"
+  import NamePanel from "./NamePanel.svelte"
+  import DataPanel from "./DataPanel.svelte"
+  import DatasourceConfigPanel from "./DatasourceConfigPanel.svelte"
+  import ExampleApp from "./ExampleApp.svelte"
+  import { FancyButton, notifications, Modal } from "@budibase/bbui"
+  import IntegrationIcon from "components/backend/DatasourceNavigator/IntegrationIcon.svelte"
+  import { SplitPage } from "@budibase/frontend-core"
+  import { API } from "api"
+  import { store, automationStore } from "builderStore"
+  import { saveDatasource } from "builderStore/datasource"
+  import { integrations } from "stores/backend"
+  import { auth, admin } from "stores/portal"
+  import FontAwesomeIcon from "components/common/FontAwesomeIcon.svelte"
+  import CreateTableModal from "components/backend/TableNavigator/modals/CreateTableModal.svelte"
+
+  let error = null
+  let name = ""
+  let url = ""
+  let stage = "name"
+  let appId = null
+
+  let plusIntegrations = {}
+  let integrationsLoading = true
+  $: getIntegrations()
+
+  let uploadModal
+
+  const createApp = async useSampleData => {
+    // Create form data to create app
+    // This is form based and not JSON
+    let data = new FormData()
+    data.append("name", name.trim())
+    data.append("url", url.trim())
+
+    if (useSampleData) {
+      data.append("sampleData", true)
+    }
+
+    const createdApp = await API.createApp(data)
+
+    // TODO I'm unsure if all this stuff is necessary, app creation seems to work without it but
```

Comment:
Anyone any context on if these methods are necessary?

Reply:
If it's the same code that runs when you normally create an app then I would keep it. I'm not entirely sure either why it's all required but I'd rather play it safe.

I agree it looks entirely redundant and we have so many wasted API calls but we don't have time to investigate it fully, so just go for whatever the normal app creation flow does.